src/refinement/loopsubdivision.jl

```diff
@@ -0,0 +1,62 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+  LoopSubdivision()
+
+Refinement of a mesh by preliminarly triangulating it if needed and
+then subdividing each triangle into four triangles.
+
+## References
+
+* Charles Loop. 1987. [Smooth subdivision surfaces based on 
+  triangles](https://charlesloop.com/thesis.pdf). 
+  Master's thesis, University of Utah.
+"""
+struct LoopSubdivision <: RefinementMethod end
+
+function refine(mesh, ::LoopSubdivision)
+  # triangulate mesh if necessary
+  tmesh = eltype(mesh) <: Triangle ? mesh : simplexify(mesh)
+
+  # retrieve initial list of points
+  points = vertices(tmesh)
+
+  # initial number of points and triangles
+  np = nvertices(tmesh)
+  nt = nelements(tmesh)
+
+  # convert topology to half-edge structure
+  t = convert(HalfEdgeTopology, topology(tmesh))
+
+  # add midpoints of edges
+  midpoints = Dict{Tuple{Int,Int},Int}()
+  ∂₁₀ = Boundary{1,0}(t)
+  for eind in 1:nfacets(t)
+    i, j = sort(∂₁₀(eind))
+    edge = Segment(points[i], points[j])
+    push!(points, center(edge))
```

Comment:
A final question: in the Loop's thesis, the subdivision procedure is similar to Catmull-Clark in the sense that the initial mesh is considered a de Boor net and the iteration process smooths it out with midpoints that are not necessarily on the edges. Perhaps we should do the same here and "smooth" the points.  Alternatively, we could find a different name to the method that is not LoopSubdivision. Which method do you prefer?

At some point in the future we need to refactor these refinement methods to give more flexibility on the way the new vertices are computed. We have a lot of code repetition currently. For example, QuadRefinement and CatmullClark are basically the same algorithm, except for the new vertex expressions.